### DISCUSSION TITLE
Button Refractor

### DISCUSSION THREAD

**Discussion by arpitBhalla** [react-native-elements/react-native-elements#3118] [Ideas]
Currently there is lots of props which can be reduced, (for example styles, `containerStyle`, `buttonStyle`, `titleStyle`, `iconContainerStyle`, `loadingStyle`) in `Button`, by doing:

 ```js
 <Button>
   <Icon ...>
   My text
 </Button>
 ```
For customizing TextStyle
 ```js
 <Button>
   <Icon ...>
   <Text style={{ ... }}>My text</Text>
 </Button>
 ```
For iconPosition we can use `orientation='vertical' | 'horizontal'` OR `display='column' | 'row'` OR Boolean `column`.

I think this will make component more customisable as users can add their own child as well, and make it easy to use, also in HTML/React the button is like `<button>My Text</button>` or in RN-Paper & NativeBase, it would be easy for user to understand & implement different libraries.

As we can't Break lib with this change, So it could be named `MaterialButton` OR `Button2`.

**Comment by dhruvdutt** [comment]
@pranshuchittora @arpitBhalla Can you add some details around why the change in this API? What's the motivation behind it and how will this help?
